### ISSUE TITLE
Compatibility with terraform 0.10.0

### ISSUE THREAD

**Issue by philsttr** [gruntwork-io/terragrunt#252]
At least one change in [terraform 0.10.0](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md#0100-beta1-june-22-2017) has broken terragrunt...

Specifically this change:

> The `terraform init` subcommand no longer takes a SOURCE argument to copy to the current directory. The behavior has been changed to match that of `plan` and `apply`, so that a configuration can be provided as an argument on the commandline while initializing the current directory. If a module needs to be copied into the current directory before initialization, it will have to be done manually.

Terragrunt uses the old two argument form of init, and terraform complains:

```
$ terragrunt plan
[terragrunt] [/foobar] 2017/07/05 16:55:38 Running command: terraform --version
[terragrunt] 2017/07/05 16:55:39 Reading Terragrunt config file at /foobar/terraform.tfvars
[terragrunt] 2017/07/05 16:55:39 Downloading Terraform configurations from git::ssh://git@githost/org/proj.git?ref=master into /tmp/terragrunt/Q9fDMTFutsTifIUWoQCicP-_WVY/b8EcM0nU_fk63G8D7qkIzC15V0Y
[terragrunt] 2017/07/05 16:55:39 Running command: terraform init -backend=false -get=false git::ssh://git@githost/org/proj.git?ref=master /tmp/terragrunt/Q9fDMTFutsTifIUWoQCicP-_WVY/b8EcM0nU_fk63G8D7qkIzC15V0Y
The init command expects at most one argument.

Usage: terraform init [options] [DIR]
```

There might be more incompatibilities, but this is the first one that I noticed.  Would be good to do some investigation into other incompatibilities.

**Comment by brikis98** [comment]
Gah, that sucks. Thanks for pointing this out. I just left a comment asking about workarounds and alternatives for this completely backwards incompatible change: https://github.com/hashicorp/terraform/pull/15032#issuecomment-313277025.

**Comment by jre21** [comment]
Since the comment doesn't seem to be attracting any attention from the devs, perhaps it would be worthwhile to open it as a new issue terraform's issue tracker?

**Comment by brikis98** [comment]
@jre21 A bug on Terraform's issue tracker sounds worthwhile!

**Comment by jre21** [comment]
https://github.com/hashicorp/terraform/issues/15655

**Comment by brikis98** [comment]
Thanks!

**Comment by jre21** [comment]
It looks like terraform 0.10.0 was released this morning with the `init -from-module` syntax included.

**Comment by robinbowes** [comment]
Guess who spent yesterday working out an AWS multi-env, multi-az terraform workflow using terragrunt and found it was broken this morning after `brew update`? :)

I'm reverting to 0.9.11 but would be good to see a release soon.

As a side note, would be good to see a Homebrew formula for terragrunt. I'll raise a separate issue for that.

R.

**Comment by bitbier** [comment]
Any update on whether or not this is going to happen soon? Currently blocked because I am missing a new built-in with the VPN connections. `aws_vpn_gateway_route_propagation` resource is only available in 0.10.0. :(

**Comment by brikis98** [comment]
We're discussing the best way to handle this in #261. There are some non-trivial things to decide. Feel free to chime in!

**Comment by bitbier** [comment]
Yeah I saw that. I'm looking at the PR now. :D

**Comment by brikis98** [comment]
Fixed by #261.
